lib/import/colorectal/providers/leeds/leeds_handler_colorectal.rb

```diff
@@ -1,648 +1,403 @@
+            genocolorectal_dup.add_gene_colorectal(gene)
+            genocolorectal_dup.add_gene_location(cdna)
+            genocolorectal_dup.add_protein_impact(protein)
+            process_exonic_variant(genocolorectal_dup, report_variants)
+            genocolorectal_dup.add_variant_class(@var_class)
+            genocolorectal_dup.add_status(@teststatus)
+            genotypes << genocolorectal_dup
+          end
+
+          def sentences_array
+            @report_variant.split(/\.\s+|and/)
+          end
+
+          def gene_path_hash
+            results = [PATHOGENIC_GENES_REGEX, GENE_PATH_REGEX].flat_map do |regexp|
+              sentences_array.map do |sentence|
+                sentence.match regexp
+              end
+            end.flatten
+            gene_path_hash = {}
+            results.each do |res|
+              gene_path_hash.merge!(res[:assocgene] => res[:pathogenic]&.strip) if res
+            end
+            gene_path_hash.each_value do |v|
+              v.gsub!('uncertain clinical significance', 'uncertain significance')
+            end
+            gene_path_hash
+          end
+
+          def process_multi_gene_abnormal(pos_genes, genocolorectal, genotypes)
+            sentences_array.each do |rep|
+              genes_found = rep.scan(COLORECTAL_GENES_REGEX).flatten.uniq
+              gene = (genes_found & pos_genes)[0]
+              next unless gene.present? && pos_genes.include?(gene)
+              next unless rep.scan(CDNA_REGEX).size.positive? ||
+                          rep.scan(EXON_VARIANT_REGEX).size.positive?
+
+              pos_genes -= [gene]
+              varclass = gene_path_hash[gene] || @var_class
+              process_abnormal_gene(rep, genocolorectal, genotypes, gene, varclass)
+            end
+
+            return if pos_genes.blank?
+
+            process_negative_genes(pos_genes, genocolorectal, genotypes)
+          end
+
+          def process_abnormal_gene(rep, genocolorectal, genotypes, gene, varclass)
+            genocolorectal_dup = genocolorectal.dup_colo
+            process_cdna_variant(genocolorectal_dup, rep)
+            process_exonic_variant(genocolorectal_dup, rep)
+            process_protein_impact(genocolorectal_dup, rep)
+            genocolorectal_dup.add_gene_colorectal(gene)
+            genocolorectal_dup.add_variant_class(varclass)
+            @teststatus = 10 if NON_PATH_VARCLASS.include? varclass
+            genocolorectal_dup.add_status(@teststatus)
+            genotypes.append(genocolorectal_dup)
+          end
+
+          def process_negative_genes(negative_genes, genocolorectal, genotypes)
+            negative_genes&.each do |gene|
+              genocolorectal_dup = genocolorectal.dup_colo
+              genocolorectal_dup.add_gene_colorectal(gene)
+              genocolorectal_dup.add_status(1)
+              genotypes << genocolorectal_dup
+            end
+          end
+
+          def allocate_genes(_record)
```

Comment:
record is not being used by the function- can this be removed?

Reply:
Done in latest commit